src/Lumper.UI/ViewModels/VtfBrowser/VtfBrowserViewModel.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Lumper.Lib.BSP.Struct;
+using Lumper.UI.Models;
+using Lumper.UI.Models.Matchers;
+using ReactiveUI;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace Lumper.UI.ViewModels.VtfBrowser;
+
+public partial class VtfBrowserViewModel : ViewModelBase
+{
+    public class Vtf : ReactiveObject
+    {
+        private Image<Rgba32>? _image;
+        private string _name = "";
+
+        public Image<Rgba32>? Image
+        {
+            get => _image;
+            set => this.RaiseAndSetIfChanged(ref _image, value);
+        }
+
+        public string Name
+        {
+            get => _name;
+            set => this.RaiseAndSetIfChanged(ref _name, value);
+        }
+    }
+
+    private double _dimensions = 128;
+
+    public double Dimensions
+    {
+        get => _dimensions;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _dimensions, value);
+            this.RaisePropertyChanged(nameof(MaxNameWidth));
+        }
+    }
+
+    public uint MaxNameWidth => (uint)_dimensions - 8;
+
+    private bool _showCubemaps;
+
+    public bool ShowCubemaps
+    {
+        get => _showCubemaps;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _showCubemaps, value);
+            this.RaisePropertyChanged(nameof(TextureBrowserItems));
+            this.RaisePropertyChanged(nameof(TexturesCount));
```

Comment:
this should be called when the list changes and not when somehing which is goning to change the list is called, right? also, does the count in UI actually change? didn't see it

Reply:
well `_showCubemaps` is used in `TextureBrowserItems` but I don't really get what you mean, and yes, the count does change
![resim](https://github.com/momentum-mod/lumper/assets/45337750/0db81171-5c36-4c9e-b2c2-24b592788ec3)
![Uploading resim.png…]()